.github/workflows/ci.yml

```diff
@@ -292,43 +292,61 @@ jobs:
       - name: Get branch and commit hash
         id: vars
         shell: bash
-        run: |
-          echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-          echo "::set-output name=branch_name::$(git rev-parse --abbrev-ref HEAD)"
-          echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
 
       - name: Build tests docker image
-        run: cd systest && make docker
+        run: make -C systest docker
 
       - name: Push tests docker images
-        run: cd systest && make push
+        run: make -C systest push
+
+      - name: set up go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.go-version }}
 
       - name: Run tests I
         timeout-minutes: 50
-        run: cd systest && make run test_name=. size=50 bootstrap=10m level=info clusters=2 node_selector=cloud.google.com/gke-nodepool=gha label=sanity storage=premium-rwo=10Gi
+        env:
+          test_id: systest-${{ steps.vars.outputs.sha_short }}
+          label: sanity
+          storage=premium-rwo: 10Gi
+          node_selector: cloud.google.com/gke-nodepool=gha
+          size: 50
+          bootstrap: 10m
+          level: info
+          clusters: 2
+        run: make -C systest run test_name=.
 
       - name: Run tests II
         timeout-minutes: 30
-        run: cd systest && make run test_name=. size=30 bootstrap=10m level=info clusters=2 node_selector=cloud.google.com/gke-nodepool=gha label=destructive storage=premium-rwo=10Gi
+        env:
+          test_id: systest-${{ steps.vars.outputs.sha_short }}
+          label: destructive
+          storage=premium-rwo: 10Gi
+          node_selector: cloud.google.com/gke-nodepool=gha
+          size: 30
+          bootstrap: 10m
+          level: info
+          clusters: 2
+        run: make -C systest run test_name=.
 
       - name: System test result
-        # SUCCESS will capture all the pods exit status separated by space. adding up all the numbers.
         run: |
-          SUCCESS=$(kubectl get pod --selector testid=systest-${{ steps.vars.outputs.sha_short }} -o json | jq .items[].status.containerStatuses[].state.terminated.exitCode)
-          SUCCESS=$(echo $SUCCESS | tr ' ' '+' | bc -l)
-          echo "::set-output name=content::$SUCCESS"
+          PASSED_TESTS=$(kubectl logs --selector testid=systest-${{ steps.vars.outputs.sha_short }} --tail=1 | grep PASS | wc -l)
+          echo "$PASSED_TESTS tests passed"
+          echo "passed_tests=$PASSED_TESTS" >> $GITHUB_OUTPUT
```

Comment:
The advantage of this previous approach was that if the number of executed tests changes the job doesn't have to be updated. Can the new approach be updated to also just check if all tests print `PASS` instead of counting the number of jobs that printed it?

Reply:
I know. I originally had it check if none test printed "FAIL" at the end. But It wouldn't work if the tests crashed (it would print neither PASS nor FAIL). Do you have any ideas how to improve it?

Reply:
The exit code of the container started by the job is overwritten with `|| exit 0`. Is this required? The test binary returns a non-0 exit code when at least one test failed.